### PR TITLE
Move CanBeExposed helper from factory_client_access

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

Moves additional helpers from the "factory_client_access". Continues work from https://github.com/kubernetes/kubernetes/pull/63984 and https://github.com/kubernetes/kubernetes/pull/63884

Moves CanBeExposed from factory_client_access.

cc @soltysh

### CODE REVIEW

File: pkg/kubectl/polymorphichelpers/helpers.go [hunk 1]
```diff
@@ -233,3 +276,162 @@ func SelectorsForObject(object runtime.Object) (namespace string, selector label
 
 	return namespace, selector, nil
 }
+
+// Generators returns the generators for the provided command
+func Generators(cmdName string) map[string]kubectl.Generator {
```
Comment (deads2k) [inline pkg/kubectl/polymorphichelpers/helpers.go]: No, these shouldn't live here.  Generators have been a disaster, but this isn't how to fix them.

File: pkg/kubectl/cmd/expose.go [hunk 2]
```diff
@@ -358,3 +363,15 @@ func (o *ExposeServiceOptions) RunExpose(cmd *cobra.Command, args []string) erro
 	}
 	return nil
 }
+
+// Check whether the kind of resources could be exposed
+func canBeExposed(kind schema.GroupKind) error {
```
Comment (deads2k) [inline pkg/kubectl/cmd/expose.go]: This one is a polymorphic helper.

Comment (juanvallejo) [reply pkg/kubectl/cmd/expose.go]: Done

**Comment by deads2k** [comment]
bazel needs updating

/lgtm

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by juanvallejo** [comment]
Rebased against latest master.
Added commits from https://github.com/kubernetes/kubernetes/pull/64121
Added commits from https://github.com/kubernetes/kubernetes/pull/64162

**Pull request merged**